ITK classes using new enum format now create runtime libraries / dependencies

### Description

Compiling an ITK remote module can result in this error being repeated multiple times.:

itk::RecursiveGaussianImageFilter<itk::Image<float, 3u>, itk::Image<float, 3u> >::PrintSelf(std::ostream&, itk::Indent) const: error: undefined reference to 'itk::operator<<(std::ostream&, itk::RecursiveGaussianImageFilterEnums::GaussianOrder)'


### Expected information

It needs to be documented that ITK classes that use enums now have both a .hxx and a .cxx file associated with them and correspondingly create shared libraries.   These libraries contain the definition of the function for streaming enum variables via << operator.

### Actual information

This change from 5.0 to 5.1 is not documented, that I could find.

### Versions

5.1rc2 and beyond

### Additional Information

Previously, this itk-module.cmake file would work:
itk_module( _YourModuleName_
  DEPENDS
    ITKCommon
    ITKStatistics
    ITKSpatialObjects
  COMPILE_DEPENDS
    ...
    ITKRegionGrowing
    ITKSmoothing
    ...

Now that file must be

itk_module( _YourModuleName_
  DEPENDS
    ITKCommon
    ITKRegionGrowing
    ITKSmoothing
    ITKStatistics
    ITKSpatialObjects
  COMPILE_DEPENDS
    ...

--- a/Documentation/ITK5MigrationGuide.md
+++ b/Documentation/ITK5MigrationGuide.md
@@ -580,6 +580,19 @@ explicit `static_cast`'s may be required in migrated code.
 Common enums used in classes across the toolkit are now available in the
 `itk::CommonEnums` class.
 
+As a consequence of improved enum support, a new library binary to provide
+enum class `operator<<` definitions is required for many ITK modules that were
+previously header-only. As a result, remote modules may encounter build-time
+link errors such as:
+
+```
+itk::RecursiveGaussianImageFilter<itk::Image<float, 3u>, itk::Image<float, 3u> >::PrintSelf(std::ostream&, itk::Indent) const: error: undefined reference to 'itk::operator<<(std::ostream&, itk::RecursiveGaussianImageFilterEnums::GaussianOrder)' `itk::CommonEnums` class.
+```
+
+To resolve these errors, update the remote module's dependency specification in
+`itk-module.cmake`. List the dependent module under `DEPENDS` instead of
+`COMPILE_DEPENDS`; this will link to the new dependent library.
+
 Update scripts
 --------------
 

